File: OOP/Lab 1/python/main.py
from Piece import Piece
class Main:
    def main():
        pieces = []

        for i in range(8):
            pieces.append(Piece("WHITE", "PAWN", i, 1))

        pieces[3].get_piece()
        pieces[3].get_coords()
        pieces[3].has_promotion_rights()

        pieces.append(Piece("BLACK", "PAWN", 5, 7))
        pieces[8].promotion()
        pieces[8].get_piece()
        pieces[8].has_promotion_rights()

if __name__ == "__main__":
    Main.main()

File: OOP/Lab 1/java/src/Main.java
import java.util.ArrayList;

// Press Shift twice to open the Search Everywhere dialog and type `show whitespaces`,
// then press Enter. You can now see whitespace characters in your code.
public class Main {
    public static void main(String[] args) {
        ArrayList<Piece> pieces = new ArrayList<Piece>();
        for(int i=0;i<8;i++){
            pieces.add(new Piece("WHITE","PAWN",i,1));
        }
        pieces.get(3).getPiece();
        pieces.get(3).getCoords();
        pieces.get(3).hasPromotionRights();

        pieces.add(new Piece("BLACK","PAWN",5,7));
        pieces.get(8).promotion();
        pieces.get(8).getPiece();
        pieces.get(8).hasPromotionRights();

    }
}